src/utils/windowUtils.ts

```diff
@@ -41,6 +41,30 @@ export module Utils {
       }
     }
 
+    /**
+     * To be used in the first line of a deprecated method
+     *
+     * @param {string} version The version when the tagged method became obsolete
+     */
+    export function deprecated(version: string) {
+
+      var callingMethod = "";
+
+      try {
+        callingMethod = (<any> new Error).stack
+          .split("\n")
+          .filter((step: string) => step.match(/http/))[1]
+          .trim()
+          .split(/\s|@/)
+          .filter((keyword: string) => !keyword.match(/at/))[0];
+      } catch (err) {
+        callingMethod = "called";
+      }
+
+      Utils.Window.warn("Method " + callingMethod + " has been deprecated in version " + version +
+        ". Please refer to the release notes.");
+    }
```

Comment:
In the future, you probably wnat to side branch this feature and then make a pull request for just `deprecated()`.  It'll make it substantially easier on the reviewer